marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -0,0 +1,250 @@
+    ev = evalValue (Slot blockNumber) inputOracles
+    in $$(interpretObservation) ev blockNumber state obs
+
+evalContract :: PubKey -> Input -> Slot
+    -> Ledger.Value -> Ledger.Value
+    -> State -> Contract
+    -> (State, Contract, Bool)
+evalContract = $$(evaluateContract)
+
+
+{-| Create Marlowe 'ValidatorScript' that remembers its owner.
+
+    At the end of a contract execution owner can spend an initial deposit
+    providing a 'Signature' for owner's public key.
+ -}
+marloweValidator :: PubKey -> ValidatorScript
+marloweValidator creator = ValidatorScript result where
+    result = Ledger.applyScript inner (Ledger.lifted creator)
+    inner  = Ledger.fromCompiledCode $$(PlutusTx.compile [|| $$(validatorScript) ||])
+
+
+{-| Create and submit a transaction that creates a Marlowe Contract @contract@
+    using @validator@ script, and put @value@ Ada as a deposit.
+-}
+createContract :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => ValidatorScript
+    -> Contract
+    -> Int
+    -> m ()
+createContract validator contract value = do
+    _ <- if value <= 0 then throwOtherError "Must contribute a positive value" else pure ()
+    slot <- slot
+    let ds = DataScript $ Ledger.lifted (Input (SpendDeposit (Signature 1)) [] [], MarloweData {
+            marloweContract = contract,
+            marloweState = emptyState })
+    let v' = Ledger.Value value
+    (payment, change) <- createPaymentWithChange v'
+    let o = scriptTxOut v' validator ds
+
+    void $ createTxAndSubmit (intervalFrom slot) payment (o : maybeToList change)
+
+
+{-| Prepare 'TxIn' and 'TxOut' generator for Marlowe style wallet actions
+-}
+marloweTx ::
+    (Input, MarloweData)
+    -> (TxOut, TxOutRef)
+    -> ValidatorScript
+    -> (TxIn -> (Int -> TxOut) -> Int -> m ())
+    -> m ()
+marloweTx inputState txOut validator f = do
+    let (TxOutOf _ (Ledger.Value contractValue) _, ref) = txOut
+    let lifted = Ledger.lifted inputState
+    let scriptIn = scriptTxIn ref validator $ Ledger.RedeemerScript lifted
+    let dataScript = DataScript lifted
+    let scritOut v = scriptTxOut (Ledger.Value v) validator dataScript
+    f scriptIn scritOut contractValue
+
+
+-- | Create Marlowe Redeemer Script as @(Input, MarloweData)@
+createRedeemer
+    :: InputCommand -> [OracleValue Int] -> [Choice] -> State -> Contract -> (Input, MarloweData)
+createRedeemer inputCommand oracles choices expectedState expectedCont =
+    let input = Input inputCommand oracles choices
+        mdata = MarloweData { marloweContract = expectedCont, marloweState = expectedState }
+    in  (input, mdata)
+
+
+{-| Create a Marlowe Commit input transaction given expected output 'Contract' and 'State'
+-}
+commit' :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => (TxOut, TxOutRef)
+    -> ValidatorScript
+    -> [OracleValue Int]
+    -> [Choice]
+    -> IdentCC
+    -> Int
+    -> State
+    -> Contract
+    -> m ()
```

Comment:
These functions with many arguments could do with Haddock comments for the arguments.